### PR TITLE
Feature: Radio mode enhancements

### PR DESCRIPTION
**Pull request by MarvinSchenkel**
This PR implements the radio mode enhancements according to https://github.com/orgs/music-assistant/projects/2?pane=issue&itemId=70609106

### CODE REVIEW

File: music_assistant/server/controllers/media/playlists.py [hunk 1]
```diff
@@ -388,47 +388,20 @@ async def _get_provider_playlist_tracks(
...
 
-    async def _get_provider_dynamic_tracks(
+    async def _get_provider_dynamic_base_tracks(
         self,
         item_id: str,
         provider_instance_id_or_domain: str,
-        limit: int = 25,
     ):
-        """Generate a dynamic list of tracks based on the playlist content."""
+        """Get the list of base tracks from the controller used to calculated the dynamic radio."""
```
Comment (marcelveldt) [inline music_assistant/server/controllers/media/playlists.py:396]:
```suggestion
        """Get the list of base tracks from the controller used to calculate the dynamic radio."""
```

File: music_assistant/server/controllers/media/radio.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ async def _update_library_item(
         await self._set_provider_mappings(db_id, provider_mappings, overwrite)
         self.logger.debug("updated %s in database: (id %s)", update.name, db_id)
 
-    async def _get_provider_dynamic_tracks(
+    async def _get_provider_dynamic_base_tracks(
```
Comment (marcelveldt) [inline music_assistant/server/controllers/media/radio.py:106]: should we remove it from the radio controller as that makes no sense ?

File: music_assistant/server/helpers/audio.py [hunk 3]
```diff
@@ -346,10 +345,6 @@ async def get_stream_details(
     if seek_position and (queue_item.media_type == MediaType.RADIO or not queue_item.duration):
         LOGGER.warning("seeking is not possible on duration-less streams!")
         seek_position = 0
-    # we use a contextvar to bypass the throttler for this asyncio task/context
-    # this makes sure that playback has priority over other requests that may be
-    # happening in the background
-    BYPASS_THROTTLER.set(True)
```
Comment (marcelveldt) [inline music_assistant/server/helpers/audio.py:352]: This may not be removed here!

**Review by marcelveldt — APPROVED** [review]
Looks good to me!
Nice work!

**Comment by OzGav** [comment]
@MarvinSchenkel is there an option or does it automatically switch to radio mode at the end of the queue per https://github.com/orgs/music-assistant/discussions/813 and the note at the end of that feature request from JLo?

**Comment by MarvinSchenkel** [comment]
> @MarvinSchenkel is there an option or does it automatically switch to radio mode at the end of the queue per https://github.com/orgs/music-assistant/discussions/813 and the note at the end of that feature request from JLo?

Not yet. First wanted to align the radio function. The "don't stop the music feature" will be added in a separate PR.

**Comment by OzGav** [comment]
@MarvinSchenkel Did you see the ERROR I posted about on Discord?

**Pull request merged**